pyfolio/timeseries.py

```diff
@@ -985,129 +985,60 @@ def rolling_sharpe(returns, rolling_sharpe_window):
         * np.sqrt(APPROX_BDAYS_PER_YEAR)
 
 
-def cone_rolling(
-        input_rets,
-        num_stdev=1.0,
-        warm_up_days_pct=0.5,
-        std_scale_factor=APPROX_BDAYS_PER_YEAR,
-        update_std_oos_rolling=False,
-        cone_fit_end_date=None,
-        extend_fit_trend=True,
-        create_future_cone=True):
-    """Computes a rolling cone to place in the cumulative returns
-    plot. See plotting.plot_rolling_returns.
+def forecast_cone_bounds(is_returns, num_days, cone_std,
+                         starting_value=1, num_samples=1000):
     """
+    Determines the upper and lower bounds of an n standard deviation
+    cone of forecasted cumulative returns. Future cumulative mean and
+    standard devation are computed by repeatedly sampling from the
+    in-sample daily returns (i.e. bootstrap). This cone is non-parametric,
+    meaning it does not assume that returns are normally distributed.
 
-    # if specifying 'cone_fit_end_date' please use a pandas compatible format,
-    # e.g. '2015-8-4', 'YYYY-MM-DD'
-
-    warm_up_days = int(warm_up_days_pct * input_rets.size)
-
-    # create initial linear fit from beginning of timeseries thru warm_up_days
-    # or the specified 'cone_fit_end_date'
-    if cone_fit_end_date is None:
-        returns = input_rets[:warm_up_days]
-    else:
-        returns = input_rets[input_rets.index < cone_fit_end_date]
-
-    perf_ts = cum_returns(returns, 1)
-
-    X = list(range(0, perf_ts.size))
-    X = sm.add_constant(X)
-    sm.OLS(perf_ts, list(range(0, len(perf_ts))))
-    line_ols = sm.OLS(perf_ts.values, X).fit()
-    fit_line_ols_coef = line_ols.params[1]
-    fit_line_ols_inter = line_ols.params[0]
-
-    x_points = list(range(0, perf_ts.size))
-    x_points = np.array(x_points) * fit_line_ols_coef + fit_line_ols_inter
+    Parameters
+    ----------
+    is_returns : pd.Series
+        In-sample daily returns of the strategy, noncumulative.
+         - See full explanation in tears.create_full_tear_sheet.
+    num_days : int
+        Number of days to project the probability cone forward.
+    cone_std : int, float, or list of int/float
+        Number of standard devations to use in the boundaries of
+        the cone. If multiple values are passed, cone bounds will
+        be generated for each value.
+    starting_value : int or float
+        Starting value of the out of sample period.
+    num_samples : int
+        Number of num_days length samples to draw from the in-sample
+        daily returns.
 
-    perf_ts_r = pd.DataFrame(perf_ts)
-    perf_ts_r.columns = ['perf']
+    Returns
+    -------
+    pd.DataFrame
+        Contains upper and lower cone boundaries. Column names are
+        strings corresponding to the number of standard devations
+        above (positive) or below (negative) the projected mean
+        cumulative returns.
+    """
+    np.random.seed(100)
```

Comment:
In the interest of producing deterministic tear sheets, might we want to set a default seed?